src/lib/address/address.ts

```diff
@@ -35,21 +58,61 @@ export const parseMaybeUrlWithChainId = ({
       try {
         const parsedUrl = parseEthUrl(urlOrAddress)
 
-        if (parsedUrl.parameters?.address) {
-          throw new Error(DANGEROUS_ETH_URL_ERROR)
+        // Handle ERC-20 token transfers
+        if (parsedUrl.parameters?.address && parsedUrl.target_address) {
+          const actualChainId = parseHexChainId(parsedUrl.chain_id, chainId)
+
+          const tokenAssetId = toAssetId({
+            chainId: actualChainId,
+            assetNamespace: ASSET_NAMESPACE.erc20,
+            assetReference: parsedUrl.target_address.toLowerCase(),
+          })
+
+          const asset = selectAssetById(store.getState(), tokenAssetId)
+          if (!asset) throw new Error(DANGEROUS_ETH_URL_ERROR)
+
+          let amountCryptoPrecision: string | undefined
+          if (parsedUrl.parameters.uint256) {
+            try {
+              amountCryptoPrecision = fromBaseUnit(parsedUrl.parameters.uint256, asset.precision)
+            } catch {
+              // Invalid amount, ignore
+            }
+          }
+
+          return {
+            assetId: tokenAssetId,
+            maybeAddress: parsedUrl.parameters.address,
+            chainId: actualChainId,
+            ...(amountCryptoPrecision && { amountCryptoPrecision }),
+          }
+        }
+
+        // Handle native ETH transfers
```

Comment:
i.e assume if it's not a token, it's a native asset - that may not be true, but from a spec point of view, it is. 
Atm, the logic has it so that if we have an asset and an amount - we skip steps all the way to amount 
We may or may not want to do that, as wallets tend to *not* have a target_address for safety, *even* if you generate a QR-code for receiving an ERC-20.
To-be-tested with a few diff wallets, buf if we indeed want to honour the intentional ambiguity, then we shouldn't set an asset here